services/frontend-service/pkg/interceptors/interceptors.go

```diff
@@ -124,6 +125,22 @@ func GoogleIAPInterceptor(
 	httpHandler(w, req)
 }
 
+func CheckPolicy(httpCtx context.Context, w http.ResponseWriter, req *http.Request, userGroup, DexRbacPolicyPath string) context.Context {
+	policy, err := auth.ReadRbacPolicy(true, DexRbacPolicyPath)
```

Comment:
the parsing (in `ReadRbacPolicy`) should happen on startup, not on a request basis.
 And if there's an error in it, we need to log the error.

Reply:
Adapted it to be read only once